src/test/crimson/test_socket.cc

```diff
@@ -313,7 +313,7 @@ class Connection {
       // successful
       logger.debug("dispatch_read_unbounded(): "
                    "expected error {}", e);
-      shutdown();
+      socket->force_shutdown_in();
```

Comment:
Thank you for the fix! This is indeed an issue, but the fix is not appropriate.

The intention of `test_unexpected_down()` is to verify if the server-side can detect an unexpected socket error (from client) and do a clean close properly, with behaviors exactly the same with `crimson::net::SocketConnection`. This means the server-side cannot use `force_shutdown_in()` or `force_shutdown_out()` and should still call `shutdown()`.

IMHO a better fix could be just treat `read-eof` as an expected case and suppress it from client. Likely at L376, change from 
```
    [] (auto cs) { return Connection::dispatch_rw_bounded(cs, 128, true); },
```
to
```
    [] (auto cs) {
      return Connection::dispatch_rw_bounded(cs, 128, true
      ).handle_exception_type([] (const std::system_error& e) {
        logger.debug("test_unexpected_down(): client got error {}", e);
        ceph_assert(e.code() == error::read_eof);
      });
    },
```

Reply:
I see. Is there another expected exception `std::errc::broken_pipe` we should consider?

Reply:
I'm not able to reproduce the case of `broken_pipe` by changing to bounded read and unbounded write from client. So currently I don't think this is an expected exception.

But that would be great if you can also reproduce the `broken_pipe` case.

Reply:
Could you also fix the typo related to "unexpected" in the PR title, commit title and message?